### PR TITLE
Rovpp virtualize and stubs

### PR DESCRIPTION
**Pull request by iReynaldo**
This completes issues 
#58: Virtualize Extrapolator functions
#64: Preserve stub ASe
#58 Create ROV Function Prototypes

I will create the proof of concept for ROV in my next pull request.

### CODE REVIEW

File: ROVppASGraph.cpp [hunk 1]
```diff
@@ -27,3 +27,11 @@ ROVppASGraph::ROVppASGraph() : ASGraph() { }
 
 ROVppASGraph::~ROVppASGraph() { }
 
+void ROVppASGraph::process(SQLQuerier *querier) {
+    // Main difference is remove_stubs isn't being called
```
Comment (c-morris) [inline ROVppASGraph.cpp]: :+1:

File: ASGraph.h [hunk 2]
```diff
@@ -45,23 +45,23 @@ class SQLQuerier;
...
+    virtual ~ASGraph();
+    // Propagation interaction
     void clear_announcements();
     uint32_t translate_asn(uint32_t asn);
     // Graph setup
     void add_relationship(uint32_t asn, uint32_t neighbor_asn, int relation);
     void process(); // Remove?
-    void process(SQLQuerier *querier); 
+    virtual void process(SQLQuerier *querier);
     void create_graph_from_db(SQLQuerier *querier);
```
[the remark this answers is not in the document]
Comment (Iuconnu) [reply ASGraph.h]: Yes, process() needs to be virtual, and should be overridden in the the ROVppAS class with the remove_stubs() function call removed. The way it is handled in the current commit is incorrect.

File: ROVppAS.cpp [hunk 3]
```diff
@@ -23,11 +23,15 @@
...
        std::set<uint32_t> *peer,
-       std::set<uint32_t> *cust) 
+       std::set<uint32_t> *cust)
     : AS(myasn, inv, prov, peer, cust)  { }
 
 ROVppAS::~ROVppAS() { }
+
+void ROVppAS::enable_rov() {
+    rov_enabled = true;
+}
```
[the remark this answers is not in the document]
Comment (Iuconnu) [reply ROVppAS.cpp]: Using 0 to 4 for the AS type flag seems simple enough. We may want a setter function that takes an int and assigns it to the AS type attribute just for clarity and simplicity.

**Comment by iReynaldo** [comment]
Okay, I'm going to cancel/close this pull request and make the suggested changes now. I'll make another pull request with the suggest changes.

**Pull request merged**